### PR TITLE
7476/mapper module

### PR DESCRIPTION
**Pull request by EVogel1999**
This PR refactors the mapper related routes into it's own module that mimics the structure of card modules.  A new shared module for access group management was also created and may need to be refactored slightly in the future (getUser function may also be a new shared function that should be in it's own module).  It also adds some new tests.  Completes story [7476/mapper module](https://app.shortcut.com/clarkcan/story/7476/mapper-module).

### CODE REVIEW

File: src/config/mongodb/MongoSingletonConnection.ts [hunk 1]
```diff
@@ -3,25 +3,34 @@ import {MongoClient, Db} from 'mongodb';
...
+export enum DATABASES {
+    ONION = 'onion',
+}
+
+export enum COLLECTIONS {
+    USERS = 'users',
+    COLLECTIONS = 'collections',
+}
 
 export class MongoSingletonConnection {
```
Comment (StarChaser44) [inline src/config/mongodb/MongoSingletonConnection.ts:35]: Add block comment for this class

**Review by mfranz1 — APPROVED** [review]

**Review by StarChaser44 — APPROVED** [review]

File: src/mapper-module/test/MapperComponent.spec.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    done();
+  });
+
+  describe('getMappers()', () => {
+    it('should resolve to get the mappers of a system', async () => {
+      await expect(component.getMappers()).resolves.toBeDefined();
+    });
+  });
+
+  describe('addMapper(memberId)', () => {
```
Comment (pzalep1) [inline src/mapper-module/test/MapperComponent.spec.ts:27]: OMG unit tests

File: src/mapper-module/test/MockAccessGroupComponent.ts [hunk 3]
```diff
@@ -0,0 +1,84 @@
+import { ResourceError, ResourceErrorReason } from "../../Error";
+import { IAccessGroupComponent } from "../../shared/modules/access-group/component/IAccessGroupComponent";
+import { AccessGroup } from "../../shared/modules/access-group/types/AccessGroups";
+
+export enum MockUserIds {
+    USER_NOT_FOUND = '0',
+    USER_TO_REMOVE_MAPPER_ACCESS = '1',
+    USER_TO_ADD_MAPPER_ACCESS = '2',
+    MAPPER_USER = '3',                  // Do not assign or change any access groups to this user
```
Comment (pzalep1) [inline src/mapper-module/test/MockAccessGroupComponent.ts:9]: I like the comments here to help designate specific mocks.

**Review by pzalep1 — APPROVED** [review]
image
Chef's kiss

**Review by JosueMolinaMorales — APPROVED** [review]

**Pull request merged**